### PR TITLE
Upgrade domain_config type in cassandra schema to add async wf config

### PR DESCRIPTION
**Pull request by taylanisikdemir**
**What changed?**
Updated `domain_config` type in cassandra schema to include async workflow config blob. This is needed to be able to persist per domain async workflow configs. The layers doing data conversion also needed to be updated.

Past relevant PRs:
- https://github.com/uber/cadence/pull/5608
- https://github.com/uber/cadence/pull/5616
- https://github.com/uber/cadence/pull/5625
- https://github.com/uber/cadence-idl/pull/158

**How did you test it?**
1. Create new image locally 
```./scripts/buildkite/docker-build.sh```

2. Run default docker compose 
```docker-compose -f docker-compose.yml up```

3. Test e2e via cli

```
# create a new domain
go run cmd/tools/cli/main.go --domain test-domain domain register --retention 2d

# get Async Queue config of the domain (should be empty)
go run cmd/tools/cli/main.go admin aq get \
  --domain test-domain

# update Async Queue config of the domain
go run cmd/tools/cli/main.go admin aq update \
  --domain test-domain \
  --json "{\"PredefinedQueueName\":\"queue1\"}"

# get Async Queue config of the domain (read what you write)
go run cmd/tools/cli/main.go admin aq get \
  --domain test-domain
```

Also tested other domain APIs such as list/describe.

### CODE REVIEW

File: common/persistence/persistence-tests/metadataPersistenceV2Test.go [hunk 1]
```diff
@@ -717,16 +717,10 @@ func (m *MetadataPersistenceSuiteV2) TestUpdateDomain() {
...
-			Topic: "topic1",
-		},
+		PredefinedQueueName: "queue1",
 	}
 	asyncWFCfg2 := types.AsyncWorkflowConfiguration{
-		QueueType: types.AsyncWorkflowQueueTypeKafka,
-		KafkaConfig: &types.AsyncWorkflowKafkaQueueConfiguration{
-			Topic: "topic2",
-		},
+		PredefinedQueueName: "queue2",
```
Comment (Shaddoll) [inline common/persistence/persistence-tests/metadataPersistenceV2Test.go:723]: Are we testing the dynamic config? I noticed that the type doesn't match our config.

Comment (taylanisikdemir) [reply common/persistence/persistence-tests/metadataPersistenceV2Test.go:723]: We will start the implementation for `PredefinedQueueName` part. Ignore the `KafkaConfig` field here. We need another round of update to replace that with a blob.

**Review by Shaddoll — APPROVED** [review]

File: scripts/buildkite/golint.sh [hunk 2]
```diff
@@ -15,5 +15,6 @@ if [ -n "$(git status --porcelain  2>&1)" ]; then
   echo "There are changes after make go-generate && make fmt && make lint && make copyright"
   echo "Please rerun the command and commit the changes"
   git status --porcelain
+  git --no-pager diff
```
Comment (davidporter-id-au) [inline scripts/buildkite/golint.sh:18]: this seem unrelated?

**Review by davidporter-id-au — APPROVED** [review]

File: scripts/buildkite/golint.sh (hunk 2 above)
Comment (taylanisikdemir) [reply scripts/buildkite/golint.sh:18]: `git status --porcelain` was showing me there were uncommitted changes but it doesn't show what those changes are to quickly address missing things. `git diff` will print those changes
